src/webview/create-component/app/component.tsx

```diff
@@ -0,0 +1,427 @@
+            appName: this.state.application.name.endsWith('-app') ? this.state.application.name : this.state.application.name + '-app'
+        });
+    }
+
+    selectFolder = (): void => {
+        VSCodeMessage.postMessage({
+            action: 'selectFolder',
+            noWSFolder: true
+        });
+    }
+
+    handleWsFolderDropDownChange = (event: any): void => {
+        if (event.target.value === 'New Folder') {
+            VSCodeMessage.postMessage({
+                action: 'selectFolder',
+                noWSFolder: true
+            });
+        } else {
+            this.setState({
+                wsFolderPath: event.target.value
+            });
+        }
+    }
+
+    handleDevFileChange = (_e: React.SyntheticEvent<Element, Event>, componentDesc: CompTypeDesc): void => {
+        this.setState({
+            selectedComponentDesc: componentDesc,
+            selectedStarterProject: componentDesc.devfileData.devfile.starterProjects ? componentDesc.devfileData.devfile.starterProjects[0].name : ''
+        })
+    }
+
+    handleStarterProjectChange = (_e: React.SyntheticEvent<Element, Event>, starterProject: string): void => {
+        this.setState({
+            selectedStarterProject: starterProject
+        })
+    }
+
+    handleCreateBtnDisable(): boolean {
+        return !this.state.component || this.state.component.name.length === 0 || this.state.component.error || this.state.application.error || !this.state.wsFolderPath ||
+            !this.state.selectedComponentDesc
+    }
+
+    includeStarterProject = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        this.setState({
+            incudeStarterProject: e.target.checked
+        })
+    }
+
+    GroupHeader = styled('div')(({ theme }) => ({
+        position: 'sticky',
+        top: '-8px',
+        padding: '4px 10px',
+        color: theme.palette.primary.main,
+        backgroundColor:
+            theme.palette.mode === 'light'
+                ? lighten(theme.palette.primary.light, 0.85)
+                : darken(theme.palette.primary.main, 0.8),
+    }));
+
+    GroupItems = styled('ul')({
+        padding: 0,
+    });
+
+    render(): React.ReactNode {
+        const { application, autoSelectDisable, component, wsFolderItems, wsFolderPath, compDescriptions, selectedComponentDesc, selectedStarterProject, showLoadScreen } = this.state;
+        const optionList = compDescriptions.sort(ascName);
+        const filterList = selectedComponentDesc ? optionList.filter((value) => value.devfileData.devfile.metadata.name === selectedComponentDesc.devfileData.devfile.metadata.name &&
+            value.registry.name === selectedComponentDesc.registry.name) : [];
+        const getInputValue = filterList.length > 0 ? filterList[0].devfileData.devfile.metadata.displayName + '/' + filterList[0].registry.name : ''
+        return (
+            <>
+                {compDescriptions.length === 0 ? <LoadScreen title={'Loading the devfiles'} /> :
```

Comment:
consider turning this into an `if` statement instead of a ternary expression in the returned value. eg.
```tsx
if (compDescriptions.length === 0) {
  return <LoadScreen .../>
} else if (showLoadScreen) {
  return <LoadScreen .../>
} else {
  return <>
    // the main form UI
  </>;
}
```